### DISCUSSION TITLE
Gateway fully confined flow

### DISCUSSION THREAD

**Discussion by liukaiyuliukaiyu** [Kong/kong#13234] [Help] [answered]
kong gateway, how to implement the full flow restriction policy when some traffic exceeds the cluster capacity, such as by request IP, by UA, etc

**Comment by jschmid1** [comment]
Hey @liukaiyuliukaiyu, you can use a combination of rate limiting and IP restriction plugins. This ensures that requests exceeding specified limits are restricted based on different criteria such as request IP, user-agent and more. 

Check our plugin hub for documentation and examples. https://docs.konghq.com/hub/

**Reply by liukaiyuliukaiyu** [reply]
rate limiting and IP restriction plugins are single route granularity, I want to implement such as limit_conn full limited flow, whether can support, no distinction between routes

**Reply by liukaiyuliukaiyu** [reply]
for Examples
server {
    limit_conn_zone $server_name zone=kong_server:500m;
    limit_conn_status 555;
    limit_conn kong_server 65000;
}

**Reply by chobits** [reply, marked as answer]
The ngx_http_limit_conn_module module is used to limit the number of connections per the defined key. As I know, there is no plugin that fullfills this requirement in Kong, so I think a good way to implement it is to add nginx directive into kong, see [this doc](https://docs.konghq.com/gateway/latest/reference/nginx-directives/#main)
